File: VB6Conv/VBA2VBProject.NET/modFileFunctions.vb
Option Strict Off
Option Explicit On
Module modFileFunctions
	Private Const BIF_RETURNONLYFSDIRS As Integer = &H1
	Private Const BIF_DONTGOBELOWDOMAIN As Integer = &H2
	Private Const BIF_RETURNFSANCESTORS As Integer = &H8
	Private Const BIF_BROWSEFORCOMPUTER As Integer = &H1000
	Private Const BIF_BROWSEFORPRINTER As Integer = &H2000
	Private Const BIF_BROWSEINCLUDEFILES As Integer = &H4000
	Private Const MAX_PATH As Integer = 260
	
	Function BrowseFolder(Optional ByRef Caption As String = "", Optional ByRef InitialFolder As String = "") As String
		
		Dim SH As Shell32.Shell
		Dim F As Shell32.Folder
		
		SH = New Shell32.Shell
		F = SH.BrowseForFolder(0, Caption, BIF_RETURNONLYFSDIRS, InitialFolder)
		If Not F Is Nothing Then
            Return F.Items.Item.Path
		End If
		
	End Function
	
	'We store our output files in a subdirectory of the original project folder
	'  so added a subfolder to the returned path.
	Public Function GetFileDirectory(ByRef strFileName As String) As String
        Dim tmpString As String
        Dim intPos As Short
		For intPos = Len(strFileName) To 1 Step -1
			If Mid(strFileName, intPos, 1) = "\" Then
                tmpString = Left(strFileName, intPos)
				Exit For
			End If
		Next intPos
        tmpString = tmpString & "VB6Conv\"
		'Dim fso As New FileSystemObject
		'UPGRADE_WARNING: Dir has a new behavior. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="9B7D5ADD-D8FE-4819-A36C-6DEDAF088CC7"'
        If Dir(tmpString, FileAttribute.Directory) = "" Then MkDir(GetFileDirectory)
        Return tmpString

	End Function
	
	Public Function GetFileNameFromPath(ByVal strFilePath As String) As String
		GetFileNameFromPath = Mid(strFilePath, InStrRev(strFilePath, "\") + 1)
	End Function
	
	Public Function StripExtensionFromFileName(ByVal strFileName As String) As String
		StripExtensionFromFileName = Left(strFileName, InStrRev(strFileName, ".") - 1)
		
	End Function
End Module

File: VB6Conv/VBA2VBProject.NET.ConvertedToC#/modFileFunctions.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.Compatibility;
using System;
using System.Collections;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Windows.Forms;
 // ERROR: Not supported in C#: OptionDeclaration
namespace VBA2VBProject
{
	static class modFileFunctions
	{
		private const int BIF_RETURNONLYFSDIRS = 0x1;
		private const int BIF_DONTGOBELOWDOMAIN = 0x2;
		private const int BIF_RETURNFSANCESTORS = 0x8;
		private const int BIF_BROWSEFORCOMPUTER = 0x1000;
		private const int BIF_BROWSEFORPRINTER = 0x2000;
		private const int BIF_BROWSEINCLUDEFILES = 0x4000;
		private const int MAX_PATH = 260;

		public static string BrowseFolder(ref string Caption = "", ref string InitialFolder = "")
		{

			Shell32.Shell SH = null;
			Shell32.Folder F = null;

			SH = new Shell32.Shell();
			F = SH.BrowseForFolder(0, Caption, BIF_RETURNONLYFSDIRS, InitialFolder);
			if ((F != null)) {
				return F.Items().Item().Path;
			}

		}

//We store our output files in a subdirectory of the original project folder
//  so added a subfolder to the returned path.
		public static string GetFileDirectory(ref string strFileName)
		{
			string tmpString = "";
			short intPos = 0;
			for (intPos = Strings.Len(strFileName); intPos >= 1; intPos += -1) {
				if (Strings.Mid(strFileName, intPos, 1) == "\\") {
					tmpString = Strings.Left(strFileName, intPos);
					break; // TODO: might not be correct. Was : Exit For
				}
			}
			tmpString = tmpString + "VB6Conv\\";
			//Dim fso As New FileSystemObject
			//UPGRADE_WARNING: Dir has a new behavior. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="9B7D5ADD-D8FE-4819-A36C-6DEDAF088CC7"'
			if (string.IsNullOrEmpty(FileSystem.Dir(tmpString, FileAttribute.Directory)))
				FileSystem.MkDir(GetFileDirectory());
			return tmpString;

		}

		public static string GetFileNameFromPath(string strFilePath)
		{
			return Strings.Mid(strFilePath, Strings.InStrRev(strFilePath, "\\") + 1);
		}

		public static string StripExtensionFromFileName(string strFileName)
		{
			return Strings.Left(strFileName, Strings.InStrRev(strFileName, ".") - 1);

		}
	}
}
